### PR TITLE
[Icons] Updating default icon theme values.

### LINKED ISSUE #3188: Default icon theme isn't using the correct color
The default icon theme isn't using the correct color (it's currently full on black, while it should be dark grey). In the image the leftmost icon should be dark grey.

Test case in: /examples/material_gallery/lib/demo/buttons_demo.dart

flutter_03

### PR DESCRIPTION
**Pull request by willlarche**
Closes #3188

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -182,9 +185,18 @@ class ThemeData extends Diagnosticable {
     hintColor ??= isDark ?  const Color(0x80FFFFFF) : const Color(0x8A000000);
     errorColor ??= Colors.red[700];
     inputDecorationTheme ??= const InputDecorationTheme();
-    iconTheme ??= isDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
-    primaryIconTheme ??= primaryIsDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
-    accentIconTheme ??= accentIsDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
+    iconTheme ??= isDark ? const IconThemeData(color: Colors.white) : const IconThemeData(
+      color: Colors.black,
+      opacity: _kIconThemeDefaultColorOpacity,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: I'm afraid that is probably overachieving. I don't think we want to start applying 0.87 to the opacity of all icons that specify a color, even though doing so is probably what's intended by the latest spec (https://material.io/design/iconography/system-icons.html#color).  

There's a fine backwards compatibility line here.  The appearance of an app that has specified an icon color explicitly shouldn't change; an app that's picking up the default icon color can change.

**Review by HansMuller — COMMENTED** [review]
- This PR should also update the default disabled color for buttons (including IconButton). Currently it's initialized like this:
```
    disabledColor ??= isDark ? Colors.white30 : Colors.black26;
```
I don't believe that's correct.
- Before committing this change we need to make sure that the resulting contrast is sufficient for a11y.
- This PR will need a tests.

**Comment by HansMuller** [comment]
It looks like MaterialButton needs a similar change:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/button.dart#L414

Should just be `theme.disabledColor`.

LIkewise for:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/button.dart#L418

And maybe for:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/switch.dart#L159

**Comment by willlarche** [comment]
What about https://github.com/flutter/flutter/blob/f353c308a11a91bae133b4edf673910020693de8/packages/flutter/lib/src/material/button.dart#L422

**Comment by HansMuller** [comment]
The button changes look good.

This issue's description should include before and after screenshots, to make it easier for developers to see what's changing.

File: packages/flutter/test/material/buttons_test.dart [hunk 2]
```diff
@@ -310,7 +310,27 @@ void main() {
...
+            onPressed: null,
+            child: Text(testText), // button is disabled
+          ),
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(buttonWidget);
+
+    final RichText text = tester.widget(find.byType(RichText));
```
Comment (HansMuller) [inline packages/flutter/test/material/buttons_test.dart]: Wouldn't `tester.widget<Text>(find.text(testText)).style` work here? The test probably doesn't need to depend on the fact that the Text widget is defined in terms of RichText.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/buttons_test.dart (hunk 2 above)
Comment (willlarche) [reply packages/flutter/test/material/buttons_test.dart]: No. You have to look for the RichText widget. The `style` on Text widgets is `null` when there's no theme. But it does draw. Just with private API to the RichText.

**Pull request merged**